seahorn/aws-c-common-stubs/s_expand_table_override.c

```diff
@@ -0,0 +1,60 @@
+#include <aws/common/hash_table.h>
+#include <aws/common/private/hash_table_impl.h>
+
+#include <seahorn/seahorn.h>
+
+#include <hash_table_helper.h>
+#include <proof_allocators.h>
+/**
+ * Function s_expand_table
+ * make non-det new_state with twice size and some empty slots
+ */
+
+int s_expand_table(struct aws_hash_table *map) {
+  struct hash_table_state *old_state = map->p_impl;
+  struct hash_table_state template = *old_state;
+
+  size_t new_size;
+  if (aws_mul_size_checked(template.size, 2, &new_size)) {
+    return AWS_OP_ERR;
+  }
+
+  if (s_update_template_size(&template, new_size)) {
+    return AWS_OP_ERR;
+  }
+
+#ifdef __SEAHORN__
+  /* Don't use s_alloc_state because that will call calloc and we want non-det
+   * values for the entries */
+  size_t required_bytes = sizeof(struct hash_table_state) +
+                          template.size * sizeof(struct hash_table_entry);
+  /* An empty slot has hashcode 0. So this marks all slots as empty */
```

Comment:
the comment is wrong, it belongs to `s_alloc_state`. We actually do not ensure that all new entries are 0 and , as the other comment says, this is intentional

Reply:
left-over comment from cbmc version, will remove

Reply:
move below ifdef. it makes sense for original code